libbeat/publisher/pipeline/monitoring.go

```diff
@@ -121,12 +123,23 @@ func (o *metricsObserver) clientClosed() { o.vars.clients.Dec() }
 func (o *metricsObserver) newEvent() {
 	o.vars.events.Inc()
 	o.vars.activeEvents.Inc()
+	o.setPercentageFull()
+}
+
+// setPercentageFull is used interally to set the `queue.full` metric
+func (o *metricsObserver) setPercentageFull() {
+	maxEvt := o.vars.queueMaxEvents.Get()
+	if maxEvt != 0 {
+		pct := float64(o.vars.activeEvents.Get()) / float64(maxEvt)
```

Comment:
You should limit the precision so we aren't including arbitrary insignificant digits. 

You could also report this as an integer between 0-100 to eliminate this problem.